doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,313 @@
+
+In this design proposal, custom tests are container images, described in the scorecard DSL yaml file.  End users would add their scorecard DSL yaml file into their operator bundle image or pass it into scorecard using a command line flag.  Custom test images are stored in a container registry outside the control of scorecard.  The scorecard DSL command flag would take precedence over the scorecard DSL config file found in the bundle image if specified.
+
+### Scorecard Configuration
+
+The scorecard DSL yaml file would be accessible from a local disk location to scorecard or from within a bundle image. 
+
+Scorecard would assume that the scorecard DSL yaml file would be located in the bundle image at the following location:
+```
+/scorecard/config.yaml
+```
+
+The scorecard DSL format is yet to be finalized but it would need to include configuration settings for each test including labels.  Here is a sample of what the scorecard DSL configuration would look like:
+```
+tests:
+# here is an example of a pair of custom tests that an ISV might implement
+- imagename: quay.io/someuser/customtest1:v0.0.1
+  description: some ISV test
+  labels:
+    suite: custom
+    test: customtest1
+- imagename: quay.io/someuser/customtest2:v0.0.1
+  description: some ISV test
+  labels:
+    test: customtest2
+# here is an example of the scorecard basic tests
+- imagename: quay.io/operator-framework/scorecard-basictests:v0.0.1
+  description: redhat - check spec
+  labels:
+    suite: basic
+    test: basic-check-spec-test
+- imagename: quay.io/operator-framework/scorecard-basictests:v0.0.1
+  description: redhat - check status
+  labels:
+    suite: basic
+    test: basic-check-status-test
+# here is an example of the scorecard olm tests
+- imagename: quay.io/operator-framework/scorecard-olmtests:v0.0.1
+  description: redhat - CRDs have validation
+  labels:
+    suite: olm
+    test: olm-crds-have-validation-test
+- imagename: quay.io/operator-framework/scorecard-olmtests:v0.0.1
+  description: redhat - CRDs have resources
+  labels:
+    suite: olm
+    test: olm-crds-have-resources-test
+- imagename: quay.io/operator-framework/scorecard-olmtests:v0.0.1
+  description: redhat - spec descriptors
+  labels:
+    suite: olm
+    test: olm-spec-descriptors-test
+- imagename: quay.io/operator-framework/scorecard-olmtests:v0.0.1
+  description: redhat - status descriptors
+  labels:
+    suite: olm
+    test: olm-status-descriptors-test
+- imagename: quay.io/operator-framework/scorecard-olmtests:v0.0.1
+  description: redhat - bundle validation
+  labels:
+    suite: olm
+    test: olm-bundle-validation-test
+# here is an example of the scorecard kuttl tests
+- imagename: quay.io/operator-framework/scorecard-kuttltests:v0.0.1
+  description: redhat - kuttl tests
+  labels:
+    suite: kuttl
+```
+
+Users could override the default test images shipped with scorecard
+by adding the following into the scorecard DSL configuration:
+```
+default-images: 
```

Comment:
I don't understand this section, if we are forcing everything in the config file.

I think CLI options to override the images are also something that we should consider.

Reply:
Agreed. IMO, the pipeline injection/override design can be deferred.

Reply:
I added a sample of what the CLI flags would look like to do an image override, I agree, having them in the config is redundant.  Fixed in the latest commit.